haystack/tools/component_tool.py

```diff
@@ -209,7 +209,7 @@ def _create_tool_parameters_schema(self, component: Component) -> Dict[str, Any]
             if socket.is_mandatory:
                 required.append(input_name)
 
-        parameters_schema = {"type": "object", "properties": properties}
+        parameters_schema = {"type": "object", "properties": properties, "additionalProperties": False}
```

Comment:
does it make sense to have it always be False? What does `True` mean here exactly?

Reply:
See https://www.apimatic.io/openapi/additionalproperties

Setting it to true only makes sense if the respective run-method accepts `**kwargs`. `**kwargs` are ignored by our input sockets anyways:
https://github.com/deepset-ai/haystack/blob/8cafcddb00cc0ab36fea842829fed091ad8062f4/haystack/core/component/component.py#L219

Reply:
sounds good and thanks for the info!